src/viam/ur/module/ur_arm.cpp

```diff
@@ -566,82 +1565,22 @@ std::vector<double> URArm::get_joint_positions(const ProtoStruct&) {
 }
 
 std::vector<double> URArm::get_joint_positions_(const std::shared_lock<std::shared_mutex>&) {
-    const std::lock_guard guard{current_state_->state_mutex};
-    if (current_state_->last_driver_status == UrDriverStatus::READ_FAILURE) {
-        // TODO(RSDK-11295): provide more context
-        throw std::runtime_error("get_joint_positions: failed to read from arm");
-    }
-
-    if (!current_state_->joints_position) {
-        // TODO(RSDK-11295): provide more context, when might this happen?
-        throw std::runtime_error("get_joint_positions: joint position is not currently known");
-    }
-
-    std::vector<double> to_ret;
-    for (const double joint_pos_rad : *current_state_->joints_position) {
-        const double joint_pos_deg = radians_to_degrees(joint_pos_rad);
-        to_ret.push_back(joint_pos_deg);
-    }
-    return to_ret;
-}
-
-std::string waypoints_filename(const std::string& path, const std::string& unix_time) {
-    constexpr char kWaypointsCsvNameTemplate[] = "/%1%_waypoints.csv";
-    auto fmt = boost::format(path + kWaypointsCsvNameTemplate);
-    return (fmt % unix_time).str();
-}
-
-std::string trajectory_filename(const std::string& path, const std::string& unix_time) {
-    constexpr char kTrajectoryCsvNameTemplate[] = "/%1%_trajectory.csv";
-    auto fmt = boost::format(path + kTrajectoryCsvNameTemplate);
-    return (fmt % unix_time).str();
-}
-
-std::string arm_joint_positions_filename(const std::string& path, const std::string& unix_time) {
-    constexpr char kArmJointPositionsCsvNameTemplate[] = "/%1%_arm_joint_positions.csv";
-    auto fmt = boost::format(path + kArmJointPositionsCsvNameTemplate);
-    return (fmt % unix_time).str();
-}
-
-std::string unix_time_iso8601() {
-    namespace chrono = std::chrono;
-    std::stringstream stream;
-
-    const auto now = chrono::system_clock::now();
-    const auto seconds_part = chrono::duration_cast<chrono::seconds>(now.time_since_epoch());
-    const auto tt = chrono::system_clock::to_time_t(chrono::system_clock::time_point{seconds_part});
-    const auto delta_us = chrono::duration_cast<chrono::microseconds>(now.time_since_epoch() - seconds_part);
-
-    struct tm buf;
-    auto* ret = gmtime_r(&tt, &buf);
-    if (ret == nullptr) {
-        throw std::runtime_error("failed to convert time to iso8601");
-    }
-    stream << std::put_time(&buf, "%FT%T");
-    stream << "." << std::setw(6) << std::setfill('0') << delta_us.count() << "Z";
-
-    return stream.str();
+    const auto radians = current_state_->read_joint_positions();
```

Comment:
@npmenard - we are going to conflict a bit here, but we basically did the same thing.